src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -4,42 +4,162 @@
  * Description: Goron in Goron Shrine / Bomb Shop Goron
  */
 
+/*
+This actor appears to be used in three scenes.
+-- Goron Shrine: Winter
+       Three variants (one instance of each used in game).
+       Main behavior determined by whether the Goron Elder's Son is asleep, and whether they themselves have been put to
+sleep. One of them also has a dialogue tree that checks whether you have lit the chandelier (at least if the kid is not
+crying)
+-- Goron Shrine: Spring
+       Six variants - one instance of each appearing in a circle around the elder on the upper level of the shrine if
+you've beaten Goht during that cycle. Dialogue determined by whether you have talked to elder in that context as a goron
+at least once during that cycle. They have a cheering animation used while the elder is talking to you, otherwise they
+stand idle.
+-- Bomb Shop
+       Only one variant. Goron who sells you powder kegs in the bomb shop.
+*/
+
 #include "z_en_s_goro.h"
+#include "../ovl_En_Gk/z_en_gk.h"
+#include "../ovl_En_Jg/z_en_jg.h"
+#include "objects/object_oF1d_map/object_oF1d_map.h"
+#include "objects/object_taisou/object_taisou.h"
 
 #define FLAGS (ACTOR_FLAG_1 | ACTOR_FLAG_8 | ACTOR_FLAG_10)
 
 #define THIS ((EnSGoro*)thisx)
 
-void EnSGoro_Init(Actor* thisx, PlayState* play);
-void EnSGoro_Destroy(Actor* thisx, PlayState* play);
-void EnSGoro_Update(Actor* thisx, PlayState* play);
-void EnSGoro_Draw(Actor* thisx, PlayState* play);
-
-#if 0
-const ActorInit En_S_Goro_InitVars = {
-    ACTOR_EN_S_GORO,
-    ACTORCAT_NPC,
-    FLAGS,
-    OBJECT_OF1D_MAP,
-    sizeof(EnSGoro),
-    (ActorFunc)EnSGoro_Init,
-    (ActorFunc)EnSGoro_Destroy,
-    (ActorFunc)EnSGoro_Update,
-    (ActorFunc)EnSGoro_Draw,
-};
+#define SET_FLAG_ON(bitfield, flag) bitfield |= flag
+#define SET_FLAG_OFF(bitfield, flag) bitfield &= ~flag
+#define FLAG_IS_SET(bitfield, flag) (bitfield & flag)
+#define CLEAR_FLAGS(bitfield) bitfield = 0U
+
+#define IS_FINAL_NIGHT (gSaveContext.save.day == 3) && (gSaveContext.save.isNight)
+
+#define ACTOR_GORON_ELDER ((EnJg*)this->otherGoron)
+#define ACTOR_GORON_CHILD ((EnGk*)this->otherGoron)
```

Comment:
```suggestion
#define ACTOR_GORON_ELDER(goron) ((EnJg*)goron)
#define ACTOR_GORON_CHILD(goron) ((EnGk*)goron)
```
I'm not 100% sure on these macros, but if we do keep them they should probably take in `goron` as an argument.